core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -114,9 +117,39 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def sortForExecutiveSummary(appsSumDesc: Seq[QualificationSummaryInfo],
       order: String): Seq[EstimatedSummaryInfo] = {
     if (QualificationArgs.isOrderAsc(order)) {
-      appsSumDesc.reverse.map(_.estimatedInfo)
+      appsSumDesc.reverse.map(sum =>
+        EstimatedSummaryInfo(
+          sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
     } else {
-      appsSumDesc.map(_.estimatedInfo)
+      appsSumDesc.map(sum => EstimatedSummaryInfo(
+        sum.estimatedInfo, sum.estimatedFrequency.getOrElse(DEFAULT_JOB_FREQUENCY)))
+    }
+  }
+
+  // Estimate app frequency based off of all applications in this run, unit jobs per month
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    val appFrequency = scala.collection.mutable.Map[String, Double]()
+    var windowStart: Long = Long.MaxValue
+    var windowEnd: Long = Long.MinValue
+
+    appsSum.foreach { sum =>
+      appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
+      windowStart = Math.min(sum.startTime, windowStart)
+      windowEnd = Math.max(windowEnd, sum.startTime + sum.estimatedInfo.appDur)
+    }
+    val windowInMonths =
+      if (windowEnd > windowStart) ((windowEnd - windowStart) / (1000.0*60*60*24*30)) else 1.0
+    // Scale frequency to per month assuming uniform distribution over the logging window rather
+    // than the individual applications window. Single run jobs are given a default frequency
+    val monthlyFrequency = appFrequency.map { case (appName, numApps) => (appName ->
+      (if (numApps <= 1) DEFAULT_JOB_FREQUENCY else (numApps / windowInMonths).round))
+    }
```

Comment:
In the above code, the timestamps (start/end) should be a map rather than a signle variable.

_Use Case_:

`windowStart` and `windowEnd` are calculated based on the entire set of applications.
Assume that:
-  `appFrequency` have two entries `{app_1: 3, app_2: 1}`
-  `windowInMonths` is positive number. let's say 15

Now, based on the calculation `monthlyFrequency`, all apps with `2 <= appFrequency < 15` will be 0, which then defaults to 1. Other apps with appFrequency will default to 3.
This means that apps with multiple entries are actually assigned to 1, while everything is 30.

The end result will be `{app_1: 1, app_1: 1, app_1: 1, app_2: 30}` 

I was expecting to see app_1 with higher frequency. Please correct me if I am wrong

Reply:
This is as intended per spec and discussions with @mattahrens -- I actually mentioned this as a follow on issue in the original PR description. Yes it seems like an anomaly, we could downscale the default frequency to 1 if the timeframe of logs is > 1 month but that's against the original specification I had.

Reply:
Sorry for the confusion.  The value of 30 is the intended default to indicate a job that only shows up once should be set to daily frequency.

Reply:
Ok, one last issue here: all the frequent apps are going to show a number smaller than 1 because the `windowEnd - windowStart` is larger than any other app-execution window. In other words, `windowInMonths` could be larger than multiple-of-month by adding the execution time of the app itself.
This could be fixed byexcluding the appDuration from windowInMonths. The calculation will be looking only to the difference between minStart and maxStart.

Reply:
Yeah, we handle the case that the frequency is less than 0.5 times per month (i.e. rounds down to 0) by forcing the minimum value to be 1 (line 152)